### PR TITLE
add needles to gradle build file

### PR DESCRIPTION
**Pull request by atomfrede**
add some needles in gradle build file for my [swagger2markup](https://github.com/atomfrede/generator-jhipster-swagger2markup) generator. With the new needles you can

* add additional build script plugins
* apply from an external gradle script
* add a new dependency

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
@atomfrede : you need to add the new functions in `modules/index.js` too

**Comment by atomfrede** [comment]
Thank @pascalgrimaud, updated the PR

**Comment by pascalgrimaud** [comment]
LGTM :+1:

File: app/templates/_build.gradle [hunk 1]
```diff
@@ -14,6 +14,7 @@ buildscript {
         classpath group: 'com.moowork.gradle', name: 'gradle-node-plugin', version: '0.11'<% if(frontendBuilder == 'grunt') {%>
         classpath group: 'com.moowork.gradle', name: 'gradle-grunt-plugin', version: '0.11'<% }  if(frontendBuilder == 'gulp') {%>
   	    classpath group: 'com.moowork.gradle', name: 'gradle-gulp-plugin', version: '0.11'<% } %>
+        //jhipster-needle-gradle-buildscript-dependency - JHipster will add additional gradle build script plugins here
```
Comment (pascalgrimaud) [inline app/templates/_build.gradle]: should be `jhipster-needle-gradle-dependency` instead ?

File: app/templates/_build.gradle [hunk 2]
```diff
@@ -199,6 +201,7 @@ dependencies {
     runtime files('lib/oracle/ojdbc/7/ojdbc-7.jar')
     runtime fileTree(dir: 'lib', include: '*.jar')<% } %>
     optional group: 'org.springframework.boot', name:'spring-boot-configuration-processor', version: spring_boot_version
+    //jhipster-needle-gradle-dependency - JHipster will add additional dependencies here
```
Comment (pascalgrimaud) [inline app/templates/_build.gradle]: should be `jhipster-needle-gradle-buildscript-dependency` instead ?

**Comment by cbornet** [comment]
We should probably have the mirror needles in maven's pom. And @atomfrede  I hope your swagger2markup module will be compatible with maven !

**Comment by pascalgrimaud** [comment]
Yes, I need these needles too : maven (todo) and gradle (already merge, thx to Fred)
But I didn't PR the needle "maven" because I try to resolve the problem about upgrading version (see the mail in dev mailing list). Maybe a system like bower, as suggested by Julien...

**Comment by atomfrede** [comment]
@pascalgrimaud Yes maven support is planned of course. Just wanted to get it up and running to see how it works.

**Pull request merged**